python/pyspark/sql/functions.py

```diff
@@ -2909,6 +2909,11 @@ def pandas_udf(f=None, returnType=None, functionType=None):
         can fail on special rows, the workaround is to incorporate the condition into the functions.
 
     .. note:: The user-defined functions do not take keyword arguments on the calling side.
+
+    .. note:: The data type of returned `pandas.Series` from the user-defined functions should be
+        matched with defined returnType. When there is mismatch between them, it is not guaranteed
+        that the conversion by SparkSQL during serialization is correct at all and users might get
```

Comment:
maybe I am concerning too much .. but how about just say .. 

```
... defined returnType (see :meth:`types.to_arrow_type` and :meth:`types.from_arrow_type`). 
When there is mismatch between them, the conversion is not guaranteed.
```

Reply:
instead of saying "conversion is not guaranteed" which sounds like results might be arbitrary, could we say "..mismatch between them, an attempt will be made to cast the data and results should be checked for accuracy."?

Reply:
>  an attempt will be made to cast the data and results should be checked for accuracy."

it sounds like the casting is intentional. I think the casting logic is not that clear as far as I can tell, comparing SQL casting logic. Can we leave this not guaranteed for now and document the casting logic here instead? Does Arrow have some kind of documentation for type conversion BTW?

Reply:
Yeah, as actually we don't intentionally cast the returned data.

How about:
```
When there is mismatch between them, Spark might do conversion on returned data.
The conversion is not guaranteed to be correct and results should be checked for accuracy by users.
```